client/signup/accordion-form/accordion-form.tsx

```diff
@@ -3,20 +3,22 @@ import { ChangeEvent, useState } from 'react';
 import AccordionFormSection from './accordion-form-section';
 import {
 	AccordionSectionProps,
-	sectionGeneratorReturnType,
+	SectionGeneratorReturnType,
 	ValidationErrors,
 	ValidatorFunction,
 } from './types';
 
 interface AccordionFormProps< T > {
-	sectionGenerator: () => (
-		props: sectionGeneratorReturnType< T >
+	generatedSections?: any[];
+	sectionGenerator?: () => (
+		props: SectionGeneratorReturnType< T >
 	) => AccordionSectionProps< T >[];
 	currentIndex: number;
 	updateCurrentIndex: ( index: number ) => void;
 	onSubmit: ( formValues: T ) => void;
 	formValuesInitialState: T;
-	updateFormValues: ( formValues: T ) => void;
+	updateFormValues?: ( formValues: T ) => void;
```

Comment:
Again in a future refactor we should probably decouple form handling logic from this accordion component. And I have also provided a call back to push up error updates as a temporary fix.